GVFS/GVFS.Platform.POSIX/ProcessLauncher.cs

```diff
@@ -54,40 +56,29 @@ public static void StartBackgroundProcess(ITracer tracer, string programName, st
         }
 
         private static unsafe void RunChildProcess(
-            ITracer tracer,
             string programName,
             byte** argvPtr = null,
             byte** envpPtr = null)
-        {
-            int fdin = Open("/dev/null", (int)NetCoreMethods.OpenFlags.O_RDONLY);
-            if (fdin == -1)
-            {
-                LogErrorAndExit(tracer, "Unable to open file descriptor for stdin", Marshal.GetLastWin32Error());
-            }
-
-            int fdout = Open("/dev/null", (int)NetCoreMethods.OpenFlags.O_WRONLY);
-            if (fdout == -1)
-            {
-                LogErrorAndExit(tracer, "Unable to open file descriptor for stdout", Marshal.GetLastWin32Error());
-            }
-
-            // Redirect stdout/stdin/stderr to "/dev/null"
-            if (Dup2(fdin, StdInFileNo) == -1 ||
-                Dup2(fdout, StdOutFileNo) == -1 ||
-                Dup2(fdout, StdErrFileNo) == -1)
-            {
-                LogErrorAndExit(tracer, "Error redirecting stdout/stdin/stderr", Marshal.GetLastWin32Error());
-            }
-
-            // Become session leader of a new session
-            if (SetSid() == -1)
+        {
+            // The daemon() function is for programs wishing to detach themselves
+            // from the controlling terminal and run in the background as system
+            // daemons.
+            //
+            // If nochdir is zero, daemon() changes the process's current working
+            // directory to the root directory("/"); otherwise, the current working
+            // directory is left unchanged.
+            //
+            // If noclose is zero, daemon() redirects standard input, standard
+            // output and standard error to /dev/ null; otherwise, no changes are
+            // made to these file descriptors.
+            if (Daemon(nochdir: 1, noclose: 0) != 0)
```

Comment:
Using Daemon is just a good general cleanup aside from the extra logging correct?  Was there any other reason for this?

Reply:
It was just to move us one step closer to standard methods/functions and away from our own custom solution.

However, this is not strictly required for the logging and so I might just take that change out for now and focus on getting us onto `Process.Start`

Reply:
@jeschu1 I decided to back this change out for now and just go with the extra logging.  I've leave the launching code as-is until we have a better fix for the hang.